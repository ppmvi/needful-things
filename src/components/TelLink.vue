<template>
  <a
    href=""
    :aria-label="`Die Nummer ${tel} anrufen`"
    @click.prevent="call"
  >
    <span
      v-if="!doNotShowTel"
      v-text="tel"
    />
    <slot />
  </a>
</template>

<script lang="ts">
  import { Vue, Component, Prop } from 'vue-property-decorator';

  @Component({
    name: 'TelLink'
  })
  export default class TelLink extends Vue {
    @Prop({ required: true }) tel!: string;
    @Prop() doNotShowTel!: boolean;

    get callableTel() {
      return this.tel.replace(/[\s\/-]/g, '').replace(/\(.*\)/g, '');
    }

    call() {
      window.location.href = `tel:${this.callableTel}`;
    }
  };
</script>
