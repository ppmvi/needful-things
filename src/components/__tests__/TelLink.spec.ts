import { mount, createLocalVue } from '@vue/test-utils';
import TelLink from '../TelLink.vue';

delete window.location;
Object.defineProperty(window, 'location', {
  value: {
    href: '',
  },
  writable: true,
});

const localVue = createLocalVue();

function wrapperFactory(props = {}) {
  return mount(TelLink, {
    localVue,
    propsData: {
      tel: '(0) 123-456 789',
      ...props,
    },
  });
}

describe('EmailLink.vue', () => {
  test('renders correctly', () => {
    const wrapper = wrapperFactory();
    expect(wrapper.element).toMatchSnapshot();
  });

  test('component is a vue instance', () => {
    const wrapper = wrapperFactory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('should not render tel when doNotShowTel is true', () => {
    const wrapper = wrapperFactory({ doNotShowTel: true });
    expect(wrapper.find('span').exists()).toBeFalsy();
  });

  test('passed tel should remove white spaces, brackets and hyphen', () => {
    const wrapper = wrapperFactory();
    expect((wrapper.vm as any).callableTel).toBe('123456789');
    wrapper.setProps({
      tel: '+49 (0) 123 456 789',
    });
    expect((wrapper.vm as any).callableTel).toBe('+49123456789');
  });

  test('click on "a" tag should set window.location.href to mailto', () => {
    const wrapper = wrapperFactory();
    wrapper.find('a').trigger('click');
    expect(window.location.href).toBe('tel:123456789');
  });
});
