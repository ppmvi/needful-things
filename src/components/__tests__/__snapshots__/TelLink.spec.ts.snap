// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailLink.vue renders correctly 1`] = `
<a
  aria-label="Die Nummer (0) 123-456 789 anrufen"
  href=""
>
  <span>
    (0) 123-456 789
  </span>
   
</a>
`;
